eth/peer.go

```diff
@@ -196,9 +196,13 @@ func (p *peer) MarkTransaction(hash common.Hash) {
 // SendTransactions sends transactions to the peer and includes the hashes
 // in its transaction hash set for future reference.
 func (p *peer) SendTransactions(txs types.Transactions) error {
+	// Mark all the transactions as known, but ensure we don't overflow our limits
 	for _, tx := range txs {
 		p.knownTxs.Add(tx.Hash())
 	}
+	for p.knownTxs.Cardinality() >= maxKnownTxs {
+		p.knownTxs.Pop()
+	}
```

Comment:
Sure we won't pop the ones we just added?

Reply:
No, but should be fine. It's generally random, worst case we double propagate something. We'll need a proper bloom version anyway long term, better keep it simple now and fix it proper later.